content/src/tnips/tnip_2.md

```diff
@@ -0,0 +1,279 @@
+
+## Motivation
+
+The foundational security of blockchain protocols relies heavily on the decentralization and unpredictability of consensus participants. A deterministic or predictable pattern in validator selection leads to vulnerabilities, such as targeted attacks and collusion, ultimately compromising the integrity and security of the network.
+
+In Ethereum, validators are randomly shuffled into different committees responsible for validating specific portions of transactions and proposing new blocks. This randomness is critical as it prevents bad actors from predicting their assignments and colluding to influence the network unfairly. By leveraging a similar random committee selection mechanism, Telcoin Network can enhance its robustness against such attacks. Randomly shuffling validators into consensus committees not only minimizes the risks of collusion, it ensures a fair representation of the validator pool and increases the difficulty for any adversarial entities aiming to control the committee selection process.
+
+Introducing a process to rotate eligible validators provides fresh opportunities for new nodes to participate in consensus as voting committee members. Randomly shuffling validators underpins both network security and community growth, aligning with the broader goals of decentralized security and equitable participation.
+
+## Specification
+##### Handshake
+
+The handshake is permissionless. [NVVs](#non-voting-validators-nvvs) initiate contact with known validators to subscribe to the latest consensus. The handshake involves exchanging network information to establish peer connections.
+
+### Committees
+
+Consensus is managed by nodes that staked TEL. These nodes form committees and cast votes to reach a quorum. Committees are only valid for the epoch in which they are current.
+
+##### Committee Voting Validators ([CVVs](#committee-voting-validators-cvvs))
+
+Validators currently in the committee, responsible for casting votes, extending the canonical chain, and reaching consensus on transactions and finality.
+
+##### Non-Voting Validators ([NVVs](#non-voting-validators-nvvs))
+
+Validators that track and execute consensus but do not vote for every block in the epoch. They participate by receiving consensus through a gossip network consisting of both [CVVs](#committee-voting-validators-cvvs) and [NVVs](#non-voting-validators-nvvs). [NVVs](#non-voting-validators-nvvs) only vote on the latest execution result to support new committees during epoch transitions.
+
+##### On-chain Committee Information
+
+Committee information is stored on-chain to support client verification. The “Consensus Registry” contract is located at 0x467Bccd9d29f223BcE8043b84E8C8B282827790F and includes the following committee types:
+
+* C: The current committee which votes to reach consensus and extend the canonical tip.
```

Comment:
Ethereum uses this approach, but I think we only need to track the previous epoch because TN requires a quorum of signatures from all staked nodes to close the epoch. This can take a while for Ethereum but is expected to be much quicker for TN

Reply:
I'm not sure if there's any additional benefit to tracking previous committees past 1, but I'll think about it some more